src/corecel/sys/Stream.cc

```diff
@@ -34,27 +35,42 @@ namespace
 void* malloc_async_impl(std::size_t bytes, Stream::StreamT s)
 {
     void* ptr{};
+    if (Stream::async())
+    {
 #if CELER_STREAM_SUPPORTS_ASYNC
-    CELER_DEVICE_CALL_PREFIX(MallocAsync(&ptr, bytes, s));
+        CELER_DEVICE_CALL_PREFIX(MallocAsync(&ptr, bytes, s));
 #else
-    CELER_DISCARD(bytes);
-    CELER_DISCARD(s);
-    CELER_DEVICE_CALL_PREFIX(Malloc(&ptr, bytes));
+        CELER_DISCARD(ptr);
+        CELER_DISCARD(bytes);
+        CELER_DISCARD(s);
+        CELER_ASSERT_UNREACHABLE();
 #endif
+    }
+    else
+    {
+        CELER_DEVICE_CALL_PREFIX(Malloc(&ptr, bytes));
+    }
     return ptr;
 }
 
 //---------------------------------------------------------------------------//
 //! Free asynchronously for CUDA and newer HIP versions
 void free_async_impl(void* ptr, Stream::StreamT s)
 {
+    if (Stream::async())
+    {
 #if CELER_STREAM_SUPPORTS_ASYNC
```

Comment:
`Stream::async` already checks `CELER_STREAM_SUPPORTS_ASYNC`so I don't think it needs to be rechecked here.

Reply:
It's necessary to disable it for HIP < 5.2 right? The API call isn't available so this would error out.

Reply:
Ah right it's still needed to prevent compile error.